README.md

```diff
@@ -433,24 +434,58 @@ const isActive = z.boolean({
 
 ## Dates
 
-z.date() accepts a date, not a date string
+Declare a date schema with the `z.date()` function.
 
 ```ts
 z.date().safeParse(new Date()); // success: true
 z.date().safeParse("2022-01-12T00:00:00.000Z"); // success: false
 ```
 
-To allow for dates or date strings, you can use preprocess
+> This schemas validate `Date` objects, not date strings. To allow for dates or
+> date strings, you can use preprocess:
+>
+> ```ts
+> const dateSchema = z.preprocess((arg) => {
+>   if (typeof arg == "string" || arg instanceof Date) return new Date(arg);
+> }, z.date());
+> type DateSchema = z.infer<typeof dateSchema>;
+> // type DateSchema = Date
+>
+> dateSchema.safeParse(new Date("1/12/22")); // success: true
+> dateSchema.safeParse("2022-01-12T00:00:00.000Z"); // success: true
+> ```
+
+## Files [browser only]
```

Comment:
Why only browser? At this point a lot of people is using Zod in browser and in Node. Why not a solution valid for both environments?

In my case it looks something like this:

``` javascript
const schema = z.object({
  name: z.string(),
  avatar: z.file() 
})

const requestHandler = (request, response) => {
  const result = parser.parse({ request, schema })

  if (!result.success) {
     // Handle error
  } 
  
  // At this point is a file in NodeJS. That can be a file in filesystem o an object in S3
  console.log("DATA", result.data.avatar)
}
```

All the implementation on how to create that file is inside the parser. But Zod is there to provide a nice validation system. Would be super nice if instead of `file` type be limited to a web `File` object would be a contract that can satisfy both web and Node.

I'm not thinking on adding any Node dependency here. Just have a file validator that accept a contract instead of a fixed `File` class. In the end the validator just want the file to respond to `size` and `type` properties right?

Reply:
I started thinking that we cannot satisfy browser and node.js in one type. My intentions were strictly to validate for File instance. I think there should be completely different type for node.js/deno and I think you should make your own in your project as handling files on server can be done in few different ways I guess.

Reply:
> I started thinking that we cannot satisfy browser and node.js in one type. My intentions were strictly to validate for File instance. I think there should be completely different type for node.js/deno and I think you should make your own in your project as handling files on server can be done in few different ways I guess.

That's the thing : ) 

1. In browser a `File` is checked for `size` and `type`
2. In Node/Deno a `Something` is checked for `size` and `type` 

The only job of this validator is to check if the `Thing` passed respond to `type` and `size`

And then in the return part could be something like this:

``` javascript
interface IFile {
  public name: string
  public size: number
  public filepath: string
}
```

I like about this proposal that is flexible enough to serve browser and Node. 

## Cons
In browser you would have to wrap your `File` with an `IFile` like interface
``` javascript
class WebFile implements IFile {
  filepath: string = ""
  constructor(file: File) {
     this.name = file.name
     this.size = file.size
  }
}
```

Reply:
Totally fine with implementing [my own file for Node](https://github.com/andresgutgon/remix-storage/blob/main/packages/bodyparser/src/zod-addons/internal/file.ts#L104) 😄 

But I think Zod is awesome and people is already using it in the server. `file` should handle both platforms somehow. But as I said totally happy if this doesn't happen inside Zod